### ISSUE TITLE
Invalid cfg generated from Mace convolution

### ISSUE THREAD

**Issue by alan-baker** [google/clspv#481]
conv_2d_3x3.cl leads to invalid cfgs.
Run with:
```
clspv -w -I . conv_2d_3x3.cl -DDATA_TYPE=float -DCMD_DATA_TYPE=f -IN_DATA_TYPE=int -DOUT_DATA_TYPE=float -DUSE_PRELU
```

Any activation should lead to the same problem.
![cfg](https://user-images.githubusercontent.com/33432579/71742225-0165a280-2e2f-11ea-8f34-1e162704deeb.png)

Clspv is generating a single block loop whose merge is the continue target of the outer loop. Validation error:
```
error: line 233: Header block 166[%166] is contained in the loop construct headed by 157[%157], but it's merge block 315[%315] is not
  %166 = OpLabel
```

**Comment by kpet** [comment]
I'm hitting this as well on a number of kernels from CLBlast.

**Comment by alan-baker** [comment]
Fixed by #485
